### PR TITLE
Feature/add three layer

### PR DESCRIPTION
**Pull request by 5Ka-me**
Added API, BLL and DAL layers.

### CODE REVIEW

File: Store/BLL/Interfaces/IProductBLL.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using BLL.Models;
+
+namespace BLL.Interfaces
+{
+    public interface IProductBLL
```
Comment (hromau) [inline Store/BLL/Interfaces/IProductBLL.cs:5]: to rename on something smooth/clean/logic

Comment (nantonov) [reply Store/BLL/Interfaces/IProductBLL.cs:5]: Better to name it like IProductService

File: Store/BLL/Models/ProductModel.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace BLL.Models
+{
+    public class ProductModel
+    {
+        public int ProductId { get; set; }
+        public string ProductName { get; set; }
+        public string ProductDescription { get; set; }
```
Comment (nantonov) [inline Store/BLL/Models/ProductModel.cs:7]: Get rid of 'Product' prefix

File: Store/DAL/Interfaces/IRepository.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using DAL.Entities;
+
+namespace DAL.Interfaces
+{
+    public interface IRepository
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:5]: Add more details to the name, Repository of what? (Product)
Rename as IProductRepository

File: Store/DAL/Interfaces/IRepository.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    {
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:6]: Get rid of access modifiers in the Interface (do not use 'public')

File: Store/DAL/Interfaces/IRepository.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        public IEnumerable<Product> GetProducts();
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:7]: Get rid of 'Products' postfix, when the interface will be named correctly - we will understand what we get

File: Store/DAL/Interfaces/IRepository.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        public Product GetProductById(int id);
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:8]: Get rid of 'Products' postfix

File: Store/DAL/Interfaces/IRepository.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+        public Product CreateProduct(Product product);
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:9]: Get rid of 'Products' postfix

File: Store/DAL/Interfaces/IRepository.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+        public Product UpdateProduct(Product product);
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:10]: Get rid of 'Products' postfix

File: Store/DAL/Interfaces/IRepository.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+        public void DeleteProduct(Product product);
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:11]: Get rid of 'Products' postfix

File: Store/DAL/Interfaces/IRepository.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+        public Product GetProductByName(string productName);
```
Comment (nantonov) [inline Store/DAL/Interfaces/IRepository.cs:12]: Get rid of 'Products' postfix

File: Store/DAL/Repository.cs [hunk 11]
```diff
@@ -0,0 +1,59 @@
+﻿using DAL.Entities;
+using DAL.Interfaces;
+using DAL.Data;
+
+namespace DAL
+{
+    public class Repository : IRepository
```
Comment (nantonov) [inline Store/DAL/Repository.cs:7]: Rename it as ProductRepository

File: Store/DAL/Interfaces/IRepository.cs (hunk 6 above)
Comment (nantonov) [reply Store/DAL/Interfaces/IRepository.cs:8]: Get rid of access modifiers for methods

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by nantonov — APPROVED** [review]

**Review by CrazyKamenwik — APPROVED** [review]

**Pull request merged**